### PR TITLE
[Protein browser] Basic protein-protein interaction table

### PR DESCRIPTION
**Pull request by rogershijin**
- Adds basic table of links to protein page below the graph
- Adds `'react-table'` package

A near-future PR will add a toggle between the two components, and another near-future PR will add table pagination.

image

### CODE REVIEW

File: static/js/biomedical/protein/types.ts [hunk 1]
```diff
@@ -32,8 +32,8 @@ export interface ProteinNode extends SimulationNodeDatum {
 
 // https://github.com/tomwanzek/d3-v4-definitelytyped/blob/06ceb1a93584083475ecb4fc8b3144f34bac6d76/src/d3-force/index.d.ts#L24
 export interface InteractionLink extends SimulationLinkDatum<ProteinNode> {
-  sourceId: string;
-  targetId: string;
+  source: string;
```
Comment (rogershijin) [inline static/js/biomedical/protein/types.ts:35]: I had messed this up originally.  (Code was still working before without type errors because SimulationLinkDatum also has source, target)

File: static/js/biomedical/protein/types.ts [hunk 2]
```diff
@@ -32,8 +32,6 @@ export interface ProteinNode extends SimulationNodeDatum {
 
 // https://github.com/tomwanzek/d3-v4-definitelytyped/blob/06ceb1a93584083475ecb4fc8b3144f34bac6d76/src/d3-force/index.d.ts#L24
 export interface InteractionLink extends SimulationLinkDatum<ProteinNode> {
-  sourceId: string;
-  targetId: string;
```
Comment (rogershijin) [inline static/js/biomedical/protein/types.ts:36]: (deleting because never used)

File: static/package.json [hunk 3]
```diff
@@ -58,9 +61,7 @@
     "typescript": "^4.7.4",
     "webpack": "^5.73.0",
     "webpack-cli": "^4.10.0",
-    "webpack-remove-empty-scripts": "^0.6.3",
-    "jszip": "^3.10.0",
-    "papaparse": "^5.3.2"
+    "webpack-remove-empty-scripts": "^0.6.3"
```
Comment (chejennifer) [inline static/package.json:64]: please revert this change, jszip and papaparse are used in other components in the repo

File: static/js/biomedical/protein/protein_protein_interaction_graph.tsx [hunk 4]
```diff
@@ -103,7 +108,26 @@ export class ProteinProteinInteractionGraph extends React.Component<
     }
     return (
       <>
-        <div id={CHART_ID}></div>
+        <div id={GRAPH_ID}></div>
+        <ProteinProteinInteractionTable
+          columns={[
```
Comment (chejennifer) [inline static/js/biomedical/protein/protein_protein_interaction_graph.tsx:113]: this looks like a constant and so probably doesn't need to be passed as a prop?

File: static/js/biomedical/protein/protein_protein_interaction_table.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+import React from "react";
+import { Column, useTable } from "react-table";
+import { Table } from "reactstrap";
+
+import { InteractionLink } from "./types";
+
+interface Props {
+  columns: readonly Column<InteractionLink>[];
+  data: readonly InteractionLink[];
```
Comment (chejennifer) [inline static/js/biomedical/protein/protein_protein_interaction_table.tsx:29]: don't need to add `readonly`

**Review by chejennifer — COMMENTED** [review]
This looks cool!

File: static/package.json (hunk 3 above)
Comment (rogershijin) [reply static/package.json:64]: Thanks for the review! Sorry I think I blindly ran some npm commands without checking the aftereffects for deletions.  Thanks for catching!

Comment (rogershijin) [reply static/package.json:64]: oh! I think jszip, papaparse are both still there (npm install might've sorted the list)

File: static/js/biomedical/protein/protein_protein_interaction_graph.tsx (hunk 4 above)
Comment (rogershijin) [reply static/js/biomedical/protein/protein_protein_interaction_graph.tsx:113]: got it will move to table component!

File: static/js/biomedical/protein/protein_protein_interaction_table.tsx (hunk 5 above)
Comment (rogershijin) [reply static/js/biomedical/protein/protein_protein_interaction_table.tsx:29]: got it will remove!

File: static/js/biomedical/protein/protein_protein_interaction_graph.tsx [hunk 6]
```diff
@@ -108,7 +113,12 @@ export class ProteinProteinInteractionGraph extends React.Component<
     }
     return (
       <>
-        <div id={CHART_ID}></div>
+        <div id={GRAPH_ID}></div>
+        <ProteinProteinInteractionTable
```
Comment (chejennifer) [inline static/js/biomedical/protein/protein_protein_interaction_graph.tsx:117]: nit: you can actually call a component like 
`<ProteinProteinInteractionTable data={...} />` and not need to add a separate closing tag

**Review by chejennifer — APPROVED** [review]
Thanks for the changes, just a small nit

File: static/js/biomedical/protein/protein_protein_interaction_graph.tsx (hunk 6 above)
Comment (rogershijin) [reply static/js/biomedical/protein/protein_protein_interaction_graph.tsx:117]: Thanks this is great syntax!
